napalm_eos/eos.py

```diff
@@ -1201,3 +1201,162 @@ def traceroute(self, destination, source='', ttl=0, timeout=0):
                 previous_probe_ip_address = ip_address
 
         return traceroute_result
+
+    def get_bgp_neighbors_detail(self, neighbor_address=''):
+        """Function to return detailed BGP information.
+
+        Information returned by this function can be about a single peer or
+        about all peers. This can be controlled using the following:
+
+        :params neighbor_address: Retuns the statistics for a specific
+                                  BGP neighbor or for all BGP neighbors.
+        """
+
+        peer_ver = None
+        is_all = True if not neighbor_address else False
+
+        if is_all:
+            command = ['show ip bgp neighbors vrf all']
+        else:
+            peer_ver = IPAddress(neighbor_address).version
+            if peer_ver == 4:
+                command = ['show ip bgp neighbors %s vrf all' %
+                           neighbor_address]
+            else:
+                command = ['show ipv6 bgp neighbors %s vrf all' %
+                           neighbor_address]
+
+        raw_output = (
+            self.device.run_commands(command, encoding='text')[0]['output'])
+
+        bgp_detail_info = {}
+        int_fields = ['accepted_prefix_count', 'local_as', 'remote_as',
+                      'local_port', 'remote_port', 'local_port',
+                      'input_messages', 'output_messages', 'input_updates',
+                      'output_updates', 'messages_queued_out', 'holdtime',
+                      'configured_holdtime', 'keepalive',
+                      'configured_keepalive', 'advertise_prefix_count',
+                      'received_prefix_count']
+
+        # Parsing through the raw bgp output
+        if peer_ver == 6:
+            neighbor_match_string = (
+                r'BGP neighbor is (.*), remote AS (\d+),')
```

Comment:
This is just to extract the list of peers from the output so as to be able to issue per peer commands later.